src/ostree/ot-builtin-create-usb.c

```diff
@@ -0,0 +1,275 @@
+                                        cancellable, error))
+      {
+        ostree_repo_abort_transaction (dest_repo, cancellable, NULL);
+        return FALSE;
+      }
+
+    if (progress != NULL)
+      ostree_async_progress_finish (progress);
+  }
+
+  /* Ensure a summary file is present to make it easier to look up commit checksums. */
+  /* FIXME: It should be possible to work without this, but find_remotes_cb() in
+   * ostree-repo-pull.c currently assumes a summary file (signed or unsigned) is
+   * present. */
+  struct stat stbuf;
+  if (!glnx_fstatat_allow_noent (ostree_repo_get_dfd (dest_repo), "summary", &stbuf, 0, error))
+    return FALSE;
+  if (errno == ENOENT &&
+      !ostree_repo_regenerate_summary (dest_repo, NULL, cancellable, error))
+    return FALSE;
+
+  /* Add the symlinks .ostree/repos.d/@symlink_name → @dest_repo_path, unless
+   * the @dest_repo_path is a well-known one like ostree/repo, in which case no
+   * symlink is necessary; #OstreeRepoFinderMount always looks there. */
+  if (!g_str_equal (dest_repo_path, "ostree/repo") &&
+      !g_str_equal (dest_repo_path, ".ostree/repo"))
+    {
+      if (!glnx_shutil_mkdir_p_at (mount_root_dfd, ".ostree/repos.d", 0755, cancellable, error))
+        return FALSE;
+
+      /* Find a unique name for the symlink. If a symlink already targets
+       * @dest_repo_path, use that and don’t create a new one. */
+      GLnxDirFdIterator repos_iter;
+      gboolean need_symlink = TRUE;
+
+      if (!glnx_dirfd_iterator_init_at (mount_root_dfd, ".ostree/repos.d", TRUE, &repos_iter, error))
+        return FALSE;
+
+      while (TRUE)
+        {
+          struct dirent *repo_dent;
+
+          if (!glnx_dirfd_iterator_next_dent (&repos_iter, &repo_dent, cancellable, error))
+            return FALSE;
+
+          if (repo_dent == NULL)
+            break;
+
+          /* Does the symlink already point to this repository? (Or is the
+           * repository itself present in repos.d?) We already guarantee that
+           * they’re on the same device. */
+          if (repo_dent->d_ino == dest_repo_stbuf.st_ino)
+            {
+              need_symlink = FALSE;
+              break;
+            }
+        }
+
+      /* If we need a symlink, find a unique name for it and create it. */
+      if (need_symlink)
+        {
+          /* Relative to .ostree/repos.d. */
+          g_autofree char *relative_dest_repo_path = g_build_filename ("..", "..", dest_repo_path, NULL);
+          guint i;
+
+          for (i = 0; i < 100; i++)
+            {
+              g_autofree char *symlink_path = g_strdup_printf (".ostree/repos.d/%02u-generated", i);
+
+              int ret = TEMP_FAILURE_RETRY (symlinkat (relative_dest_repo_path, mount_root_dfd, symlink_path));
+              if (ret < 0 && errno != EEXIST)
+                return glnx_throw_errno_prefix (error, "symlinkat(%s → %s)", symlink_path, relative_dest_repo_path);
+              else if (ret >= 0)
+                break;
```

Comment:
There's a private `_ostree_make_temporary_symlink_at()` though it's harder to use private bits from the cmdline.  It uses `glnx_gen_temp_name()` which is visible here.

Reply:
Looks fairly similar to what I’ve got in `ot-builtin-create-usb.c`. Is there anything wrong with the code I’ve got already? I’d prefer to keep the template as `%02u` since that’s a bit more human-friendly than `XXXXXX`.

Reply:
It's fine as is, just noting.